cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteServiceManager.java

```diff
@@ -74,20 +77,31 @@ public int getProvisionedInstances() {
 
   @Override
   public URL getServiceURL() {
+    return getServiceURL(30, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public URL getServiceURL(long timeout, TimeUnit timeoutUnit) {
     try {
+      Tasks.waitFor(true, new Callable<Boolean>() {
+        @Override
+        public Boolean call() throws Exception {
+          try {
+            serviceClient.checkAvailability(serviceId);
+            return true;
+          } catch (ServiceUnavailableException e) {
+            // simply retry in case its not yet available
+            return false;
+          }
+        }
+      }, timeout, timeoutUnit);
       return serviceClient.getServiceURL(serviceId);
```

Comment:
Right, it can be changed to use `ServiceId`. At that time, I think it was because `RemoteServiceManager` only stores `ProgramId` and `Id.Service serviceId`